main.tf

```diff
@@ -28,23 +28,32 @@ module "eks_tags" {
   tags        = local.tags
 }
 
+# Create the cluster's KMS key
+module "kms" {
+  source = "./modules/aws-kms"
+
+  alias                   = "alias/${module.eks_tags.id}"
+  description             = "${module.eks_tags.id} EKS cluster secret encryption key"
+  deletion_window_in_days = 30
+  enable_key_rotation     = true
+  policy                  = var.kms_key_policy
```

Comment:
If the policy is not provided by the customer then we can create opinionated kms policy for EKS Cluster.

Reply:
`try(var. kms_key_policy, data.aws_iam_policy_document.eks_key.json)`